micro-rdk/src/common/grpc.rs

```diff
@@ -1276,6 +1287,72 @@ impl<'a> GrpcServerInner<'a> {
     }
 
+    fn switch_set_position(&mut self, message: &[u8]) -> Result<Bytes, ServerError> {
+        let req = component::switch::v1::SetPositionRequest::decode(message)
+            .map_err(|_| ServerError::from(GrpcError::RpcInvalidArgument))?;
+        let switch = match self.robot.lock().unwrap().get_switch_by_name(req.name) {
+            Some(m) => m,
+            None => return Err(ServerError::from(GrpcError::RpcUnavailable)),
+        };
+        switch
+            .lock()
+            .unwrap()
+            .set_position(req.position)
+            .map_err(|err| ServerError::new(GrpcError::RpcInternal, Some(err.into())))?;
+        let resp = component::switch::v1::SetPositionResponse {};
+        GrpcServerInner::encode_message(resp)
+    }
+
+    fn switch_get_position(&mut self, message: &[u8]) -> Result<Bytes, ServerError> {
+        let req = component::switch::v1::GetPositionRequest::decode(message)
+            .map_err(|_| ServerError::from(GrpcError::RpcInvalidArgument))?;
+        let switch = match self.robot.lock().unwrap().get_switch_by_name(req.name) {
+            Some(m) => m,
+            None => return Err(ServerError::from(GrpcError::RpcUnavailable)),
+        };
+        let position = switch
+            .lock()
+            .unwrap()
+            .get_position()
+            .map_err(|_| ServerError::from(GrpcError::RpcInvalidArgument))?;
+        let resp = component::switch::v1::GetPositionResponse { position };
+        GrpcServerInner::encode_message(resp)
+    }
+
+    fn switch_get_num_positions(&mut self, message: &[u8]) -> Result<Bytes, ServerError> {
+        let req = component::switch::v1::GetNumberOfPositionsRequest::decode(message)
+            .map_err(|_| ServerError::from(GrpcError::RpcInvalidArgument))?;
+        let switch = match self.robot.lock().unwrap().get_switch_by_name(req.name) {
+            Some(m) => m,
+            None => return Err(ServerError::from(GrpcError::RpcUnavailable)),
+        };
+        let number_of_positions = switch
+            .lock()
+            .unwrap()
+            .get_num_positions()
+            .map_err(|_| ServerError::from(GrpcError::RpcInvalidArgument))?;
+        let resp = component::switch::v1::GetNumberOfPositionsResponse {
+            number_of_positions,
+        };
+        GrpcServerInner::encode_message(resp)
+    }
+
+    fn switch_do_command(&mut self, message: &[u8]) -> Result<Bytes, ServerError> {
+        let req = proto::common::v1::DoCommandRequest::decode(message)
+            .map_err(|_| ServerError::from(GrpcError::RpcInvalidArgument))?;
+        let switch = match self.robot.lock().unwrap().get_switch_by_name(req.name) {
+            Some(m) => m,
+            None => return Err(ServerError::from(GrpcError::RpcUnavailable)),
+        };
+        let res = switch
+            .lock()
+            .unwrap()
+            .do_command(req.command)
+            .map_err(|_| ServerError::from(GrpcError::RpcInvalidArgument))?;
```

Comment:
here as well

Reply:
changed it, however many of the do_commands use this pattern. made a cleanup ticket [here](https://viam.atlassian.net/browse/RSDK-10504)